zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -1126,6 +1129,7 @@ function ParagraphCtrl($scope, $rootScope, $route, $window, $routeParams, $locat
         $scope.$on('editorSetting', function(event, data) {
           if (paragraph.id === data.paragraphId) {
             deferred.resolve(data);
+            $scope.editor.setReadOnly(!$scope.userHasWritePermission());
```

Comment:
I worry this would leave it in an inconsistent state - UI showing the changed value, but it isn't saved

Reply:
$scope.editor.setReadOnly(...) - blocks user input completely.
Or do you mean something else?